compiler/src/main/java/com/juul/pommel/compiler/SoloModuleGenerator.kt

```diff
@@ -0,0 +1,19 @@
+package com.juul.pommel.compiler
+
+import com.squareup.javapoet.JavaFile
+import javax.annotation.processing.Messager
+import javax.lang.model.element.Element
+
+internal interface SoloModuleGenerator {
+    fun isGeneratorFor(element: Element): Boolean
+    fun generate(soloModuleParams: PommelModule, element: Element): JavaFile
+    fun validate(element: Element, messager: Messager): PommelModule?
+}
+
+private fun generators() = sequence {
+    yield(ClassSoloModuleGenerator())
+    yield(FunctionSoloModuleGenerator())
+}
```

Comment:
This pattern was taken from [Exercise](https://github.com/JuulLabs/exercise/blob/ce5d03ce8f854c522beff96368d99bc471cee8d2/compile/src/main/java/com/juul/exercise/compile/generator/file/FileGenerator.kt#L12). I did this based on the thought that I might want conditional logic to exclude generators based on compiler flags or something, but I never ended up using that, so you're totally right that it can be simplified.